### PR TITLE
liblineage: Unbreak it (again)

### PR DESCRIPTION
**Pull request by ItsVixano**
* Following the changes on https://github.com/LineageOS/lineage_wiki/commit/858d862ad4127a5c2928c87a287748657adbd7b0

Change-Id: Id1cd2f06c600f0afca8b0e1f898c051a5fecfdc1

### CODE REVIEW

File: liblineage/wiki/device_data.py [hunk 1]
```diff
@@ -83,7 +84,7 @@ def __init__(self,
 	             gpu: str,
 	             image: str,
 	             install_method: str,
-	             kernel: str,
+	             kernel: Union[KernelData],
```
Comment (SebaUbuntu) [inline liblineage/wiki/device_data.py:87]: Optional[KernelData]?

**Review by SebaUbuntu — CHANGES_REQUESTED** [review]

File: liblineage/wiki/device_data.py (hunk 1 above)
Comment (ItsVixano) [reply liblineage/wiki/device_data.py:87]: Done

File: liblineage/wiki/device_data.py [hunk 2]
```diff
@@ -36,7 +37,7 @@ class DeviceData:
 	- gpu: GPU name
 	- image: The image of the device
 	- install_method: The install method of the device
-	- kernel: Kernel repository
+	- kernel: Kernel repository & Kernel version
```
Comment (SebaUbuntu) [inline liblineage/wiki/device_data.py]: `kernel: Kernel info`

File: liblineage/wiki/device_data.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 from datetime import date
 import requests
-from typing import Any, Dict, List, Sequence, Union
+from typing import Any, Dict, Optional, List, Sequence, Union
```
Comment (SebaUbuntu) [inline liblineage/wiki/device_data.py]: Alphabetical order

File: liblineage/wiki/device_data.py [hunk 4]
```diff
@@ -19,6 +19,7 @@
 from liblineage.wiki.data_types.release_data import ReleaseData
 from liblineage.wiki.data_types.screen_data import ScreenData
 from liblineage.wiki.data_types.sdcard_data import SdcardData
+from liblineage.wiki.data_types.kernel_data import KernelData
```
Comment (SebaUbuntu) [inline liblineage/wiki/device_data.py]: Alphabetical order

File: liblineage/wiki/device_data.py [hunk 5]
```diff
@@ -232,7 +233,8 @@ def __str__(self) -> str:
 			"CPU frequency": self._print_data(self.cpu_freq),
 			"Dimensions": self._print_data(self.dimensions),
 			"GPU": self._print_data(self.gpu),
-			"Kernel repository": f"{GITHUB_ORG_URL}/{self.kernel}",
+			"Kernel repository": f"{GITHUB_ORG_URL}/{self.kernel.repo}",
```
Comment (SebaUbuntu) [inline liblineage/wiki/device_data.py]: Use `self._print_data`, and use only one line (do it like the Bluetooth one)

**Review by SebaUbuntu — CHANGES_REQUESTED** [review]

File: liblineage/wiki/device_data.py (hunk 5 above)
Comment (ItsVixano) [reply liblineage/wiki/device_data.py]: Done

**Review by SebaUbuntu — APPROVED** [review]

File: liblineage/wiki/device_data.py [hunk 6]
```diff
@@ -6,7 +6,7 @@
 
 from datetime import date
 import requests
-from typing import Any, Dict, List, Sequence, Union
+from typing import Any, Dict, List, Optional, Sequence, Union
```
Comment (SebaUbuntu) [inline liblineage/wiki/device_data.py]: Optional isn't required anymore

File: liblineage/wiki/data_types/kernel_data.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+#
+# Copyright (C) 2023 The LineageOS Project
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+#
+
+from typing import Dict, Optional, Union
```
Comment (SebaUbuntu) [inline liblineage/wiki/data_types/kernel_data.py]: Optional isn"t used?

**Review by SebaUbuntu — CHANGES_REQUESTED** [review]

**Review by SebaUbuntu — APPROVED** [review]

**Pull request merged**